api/v1alpha1/runnerpool_types.go

```diff
@@ -57,6 +57,10 @@ type SlackAgentConfig struct {
 }
 
 type RunnerPodTemplateSec struct {
+	// Standard object's metadata.  Only `annotations` and `labels` are valid.
+	// +optional
+	ObjectMeta `json:"metadata,omitempty"`
```

Comment:
`,omitempty` here is meaningless.

Reply:
Thanks for your comment!
Why is that?
I thought that optional fields need `omitempty`, am I wrong?

Reply:
omitempty is used for JSON serialization, whereas `+optional` declares the filed is optional as OpenAPI schema.
Do you see?

They are different things, and struct (not a pointer) has no `empty` condition, so `omitempty` is meaningless.

Reply:
For reference, read https://pkg.go.dev/encoding/json#Marshal

> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.

Reply:
If you don't see that, consider that a user can send arbitrary JSON data to kube-apiserver
w/o serializing Go structs.

`+optional` is not related to Go and serialization actually. It is a validation rule for kube-apiserver.

Reply:
Thank you for the explanation.
I understand the need for `omitempty`.
I will fix it.